Engine/Generic/SuppressedRecord.cs

```diff
@@ -11,20 +13,28 @@ public class SuppressedRecord : DiagnosticRecord
         /// <summary>
         /// The rule suppression of this record
         /// </summary>
-        public RuleSuppression Suppression
+        public IList<RuleSuppression> Suppressions
```

Comment:
This should be immutable, so an `IReadOnlyList<RuleSuppression>`, and ideally shouldn't be settable. Instead it should just have the getter and have the value set through the constructor. Also because it's PowerShell exposed, we should take whatever value is given in the constructor and copy it into a ReadOnlyCollection or some other concrete readonly list implementation, so that PowerShell doesn't see through the interface and allow mutation